packages/amplify-category-auth/src/provider-utils/awscloudformation/import/ensure-headless-parameters.ts

```diff
@@ -0,0 +1,49 @@
+import { AmplifyError } from 'amplify-cli-core';
+import { ResourceParameters, ImportAuthHeadlessParameters, EnvSpecificResourceParameters } from './types';
+
+export const ensureHeadlessParameters = (
+    resourceParameters: ResourceParameters,
+    headlessParams: ImportAuthHeadlessParameters,
+  ): EnvSpecificResourceParameters => {
+    // If we are doing headless mode, validate parameter presence and overwrite the input values from env specific params since they can be
+    // different for the current env operation (eg region can mismatch)
+  
+    // Validate required arguments to be present
+    const missingParams = [];
+  
+    if (!headlessParams.userPoolId) {
+      missingParams.push('userPoolId');
+    }
+  
+    if (!headlessParams.webClientId) {
+      missingParams.push('webClientId');
+    }
+  
+    if (!headlessParams.nativeClientId) {
+      missingParams.push('nativeClientId');
+    }
+  
+    if (resourceParameters.authSelections === 'identityPoolAndUserPool' && !headlessParams.identityPoolId) {
+      missingParams.push('identityPoolId');
+    }
+  
+    if (missingParams.length > 0) {
+      throw new AmplifyError( 'InputValidationError', {
+        message: `auth headless is missing the following inputParameters ${missingParams.join(', ')}`,
+        link: 'https://docs.amplify.aws/cli/usage/headless/#--categories'
+      }); 
+    }
+  
+    const envSpecificParameters: EnvSpecificResourceParameters = {
+      userPoolId: headlessParams.userPoolId,
+      userPoolName: '', // Will be filled out later
```

Comment:
is it mandatory parameter ?

Reply:
It is not , that why I guess will be filled out later :)
May be used here  to share the Type def

Reply:
should be typed as `userPoolName?: string;` but that's out of scope of this PR, unless you want a challenge ;-)